### PR TITLE
validate cloudapp hostname using location and fix validation logic error

### LINKED ISSUE #534: health check validation for fqdn tied to region
I created a cluster where the input fqdn was `osatestshrcluster03-0928.eastus.cloudapp.azure.com` and location for cluster was `westcentralus`. This cluster creation failed in health check with error
`Health check failed. error: Get https://osatestshrcluster02-0928.eastus.cloudapp.azure.com/healthz: dial tcp: lookup osatestshrcluster02-0928.eastus.cloudapp.azure.com on 10.0.0.10:53: no such host`.

On looking closer, I could see that `https://osatestshr02cluster0928.westcentralus.cloudapp.azure.com/healthz` was up, even though the input was actually `osatestshrcluster03-0928.eastus.cloudapp.azure.com`

### PR DESCRIPTION
**Pull request by jim-minter**
fixes #534 

@pweil- @mjudeikis @kargakis ptal.

@amanohar and I believe that we should backport this.

### CODE REVIEW

File: pkg/api/validate.go [hunk 1]
```diff
@@ -265,7 +268,7 @@ func validateRouterProfile(rp RouterProfile) (errs []error) {
 		errs = append(errs, fmt.Errorf("invalid properties.routerProfiles[%q].publicSubdomain %q", rp.Name, rp.PublicSubdomain))
 	}
 
-	if rp.FQDN != "" && !isValidHostname(rp.FQDN) && !isAzureZone(rp.FQDN) {
+	if rp.FQDN != "" && !isValidCloudAppHostname(rp.FQDN, location) {
```
Comment (jim-minter) [inline pkg/api/validate.go]: pernicious! this should originally have been `if rp.FQDN != "" && (!isValidHostname(rp.FQDN) || !isAzureZone(rp.FQDN)) {`

Comment (jim-minter) [reply pkg/api/validate.go]: @thanasisk fyi ^ (also my bad for lgtm'ing it originally)

**Comment by amanohar** [comment]
lgtm

File: pkg/api/validate.go [hunk 2]
```diff
@@ -40,8 +40,11 @@ func isValidHostname(h string) bool {
 	return len(h) <= 255 && rxRfc1123.MatchString(h)
 }
 
-func isAzureZone(fqdn string) bool {
-	return strings.HasSuffix(fqdn, ".cloudapp.azure.com") && len(strings.Split(fqdn, ".")) == 5
+func isValidCloudAppHostname(h, location string) bool {
```
Comment (kargakis) [inline pkg/api/validate.go]: s/h/fqdn/

**Comment by kargakis** [comment]
just a nit

/hold cancel

**Pull request merged**